cli/pcluster/pcluster.py

```diff
@@ -595,12 +626,25 @@ def command(args, extra_args):
         stack_result = cfn.describe_stacks(StackName=stack).get("Stacks")[0]
         status = stack_result.get("StackStatus")
         valid_status = ["CREATE_COMPLETE", "UPDATE_COMPLETE", "UPDATE_ROLLBACK_COMPLETE"]
-        if status not in valid_status:
-            LOGGER.info("Stack status: %s. Stack needs to be in %s", status, " or ".join(valid_status))
+        invalid_status = ['DELETE_COMPLETE', 'DELETE_IN_PROGRESS']
+
+        if status in invalid_status:
+            LOGGER.info("Stack status: %s. Cannot SSH while in %s" % (status, ' or '.join(invalid_status)))
             sys.exit(1)
-        outputs = stack_result.get("Outputs")
-        username = [o.get("OutputValue") for o in outputs if o.get("OutputKey") == "ClusterUser"][0]
-        ip = [o.get("OutputValue") for o in outputs if o.get("OutputKey") == "MasterPublicIP"][0]
+        elif status in valid_status:
+            outputs = stack_result.get("Outputs")
+            username = [o.get("OutputValue") for o in outputs if o.get("OutputKey") == "ClusterUser"][0]
+
+            try:
+                ip = [o.get("OutputValue") for o in outputs if o.get("OutputKey") == "MasterPublicIP"][0]
+            except IndexError:
+                # If they're not using public ip's use the private ip instead
+                ip = [o.get("OutputValue") for o in outputs if o.get("OutputKey") == "MasterPrivateIP"][0]
+        else:
```

Comment:
Could you add a comment for the `else` block?

Reply:
sure